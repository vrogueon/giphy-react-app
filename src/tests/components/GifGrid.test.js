import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme/build';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid/>', () => {
    
    const category = 'jjba';
   
    test('debe de hacer match con el snapshot de la primer carga', () => {
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={ category }/>);    
        expect(wrapper).toMatchSnapshot();

    });

    test('debe de mostrar items cuando se cargan imagenes en useFetchGifs', () => {
        
        const gifs = [{
            id: 'Test123',
            url: 'https://localhost/test.jpg',
            title: 'Test test test'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: true
        });

        const wrapper = shallow(<GifGrid category={ category }/>);    
        
        expect( wrapper.find('p').exists() ).toBe(true);
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
    });
    
});
