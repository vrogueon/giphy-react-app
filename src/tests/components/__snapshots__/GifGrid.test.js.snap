// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid/> debe de hacer match con el snapshot de la primer carga 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    jjba
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
